rust/core-lib/src/movement.rs

```diff
@@ -49,6 +49,8 @@ pub enum Movement {
     StartOfParagraph,
     /// Move to the end of the text line.
     EndOfParagraph,
+    /// Move to the end of the text line, or next line if already at end.
+    EndOfParagraphKill,
```

Comment:
Is this about accommodating cocoa's weird two-selector behaviour? If so I have a PR to reopen..